portlayer/attach/connector.go

```diff
@@ -167,21 +175,30 @@ func (c *Connector) processIncoming(conn net.Conn) {
 	}
 
 	log.Debugf("Initiating ssh handshake with new connection attempt")
-	ccon, newchan, request, err := ssh.NewClientConn(conn, "", config)
+	var (
+		ccon    ssh.Conn
+		newchan <-chan ssh.NewChannel
+		request <-chan *ssh.Request
+	)
+
+	ccon, newchan, request, err = ssh.NewClientConn(conn, "", config)
 	if err != nil {
 		log.Errorf("SSH connection could not be established: %s", errors.ErrorStack(err))
 		return
 	}
 
 	client := ssh.NewClient(ccon, newchan, request)
-	ids, err := SSHls(client)
+
+	var ids []string
+	ids, err = SSHls(client)
 	if err != nil {
 		log.Errorf("SSH connection could not be established: %s", errors.ErrorStack(err))
 		return
 	}
 
 	for _, id := range ids {
-		si, err := SSHAttach(client, id)
+		var si SessionInteraction
```

Comment:
not sure what is the expected length of ids but maybe you may want to consider moving the declaration out of the loop so that we create one SessionInteraction and reuse multiple times.